Mathlib/Order/Interval/Finset/Nat.lean

```diff
@@ -101,42 +101,51 @@ theorem card_fintypeIic : Fintype.card (Set.Iic b) = b + 1 := by
 
 theorem card_fintypeIio : Fintype.card (Set.Iio b) = b := by rw [Fintype.card_ofFinset, card_Iio]
 
--- TODO@Yaël: Generalize all the following lemmas to `SuccOrder`
+@[deprecated "Use `Finset.Icc_succ_left_eq_Ioc`" (since := "2025-02-25")]
 theorem Icc_succ_left : Icc a.succ b = Ioc a b := by
   ext x
   rw [mem_Icc, mem_Ioc, succ_le_iff]
 
+@[deprecated "Use `Finset.Ico_succ_right_eq_Icc`" (since := "2025-02-25")]
 theorem Ico_succ_right : Ico a b.succ = Icc a b := by
   ext x
   rw [mem_Ico, mem_Icc, Nat.lt_succ_iff]
 
+@[deprecated "Use `Finset.Ico_succ_left_eq_Ioo`" (since := "2025-02-25")]
 theorem Ico_succ_left : Ico a.succ b = Ioo a b := by
   ext x
   rw [mem_Ico, mem_Ioo, succ_le_iff]
 
+@[deprecated "Use `Finset.Icc_pred_right_eq_Ico`" (since := "2025-02-25")]
 theorem Icc_pred_right {b : ℕ} (h : 0 < b) : Icc a (b - 1) = Ico a b := by
   ext x
   rw [mem_Icc, mem_Ico, lt_iff_le_pred h]
 
+@[deprecated "Use `Finset.Ico_succ_succ_eq_Ioc`" (since := "2025-02-25")]
 theorem Ico_succ_succ : Ico a.succ b.succ = Ioc a b := by
   ext x
   rw [mem_Ico, mem_Ioc, succ_le_iff, Nat.lt_succ_iff]
 
+set_option linter.deprecated false in
 @[simp]
 theorem Ico_succ_singleton : Ico a (a + 1) = {a} := by rw [Ico_succ_right, Icc_self]
 
+set_option linter.deprecated false in
 @[simp]
 theorem Ico_pred_singleton {a : ℕ} (h : 0 < a) : Ico (a - 1) a = {a - 1} := by
   rw [← Icc_pred_right _ h, Icc_self]
 
+set_option linter.deprecated false in
 @[simp]
 theorem Ioc_succ_singleton : Ioc b (b + 1) = {b + 1} := by rw [← Nat.Icc_succ_left, Icc_self]
 
 variable {a b c}
 
+set_option linter.deprecated false in
 theorem Ico_succ_right_eq_insert_Ico (h : a ≤ b) : Ico a (b + 1) = insert b (Ico a b) := by
```

Comment:
The naming here is pretty confusing imo: some lemmas with `+ 1` are now written as "succ" and others as "add_one". A user shouldn't need to think "does this lemma hold for SuccAddOrder or not" to figure out what to search for the name... Compare this statement with `Ico_add_one_right_eq_Icc`, where the LHS is visually spelt exactly the same, but the symbol-read is different.

Reply:
I agree, but note that I am not generalising any lemma about `insert` in this PR. I was thinking of making that a second PR, unless you insist on generalising all lemmas here at once?

Reply:
I'm happy for it to be a second PR, but let's add a new TODO for this series. Presumably on line 128?

Reply:
I've added a TODO in the two module doc